src/AaveV3StrategicAssets_20230606/VeTokenManager.sol

```diff
@@ -0,0 +1,261 @@
+  event Unlock(uint256 tokensUnlocked);
+  event VoteCast(uint256 voteData, bool support);
+  event VotingContractUpdate(address indexed token, address voting);
+
+  /// @notice The new lock duration must be greater than the current one
+  error InvalidNewDuration();
+  /// @notice The on-chain voting contract has not been set for this token
+  error VotingContractNotRegistered();
+
+  struct VeToken {
+    /// @notice the keccak encoded spaceId of the snapshot space
+    bytes32 spaceId;
+    /// @notice The lock duration of veToken
+    uint256 lockDuration;
+    /// @notice Address of veToken
+    address veToken;
+    /// @notice the snapshot delegate for the deposit
+    address delegate;
+    /// @notice Warden contract address for veBoost
+    address warden;
+  }
+
+  DelegateRegistry public constant DELEGATE_REGISTRY =
+    DelegateRegistry(0x469788fE6E9E9681C6ebF3bF78e7Fd26Fc015446);
+
+  mapping(address => VeToken) public veTokens;
+  mapping(address => address) public veTokenToVotingContract;
+
+  /// @notice The Buyer will then need to call the user_checkpoint() method on all Curve V4 & Factory Gauges
+  /// currently farming, to apply the newly received Boost.
+  /// @notice duration in weeks (ie: 1 for 1 Week)
+  function buyBoost(
+    address underlying,
+    address delegator,
+    address receiver,
+    uint256 amount,
+    uint256 duration
+  ) external onlyOwnerOrGuardian {
+    VeToken storage token = veTokens[underlying];
+    if (token.warden == address(0)) revert TokenNotRegistered();
+
+    IERC20 feeToken = IWardenBoost(token.warden).feeToken();
+    uint256 maxFee = IWardenBoost(token.warden).estimateFees(delegator, amount, duration);
+    feeToken.approve(token.warden, maxFee);
+    IWardenBoost(token.warden).buyDelegationBoost(delegator, receiver, amount, duration, maxFee);
+
+    emit BuyBoost(delegator, receiver, amount, duration);
+  }
+
+  /// @notice Sells boost in order to accrue rewards
+  /// @dev Registers a new user, creates a BoostOffer with the given parameters
+  /// @param pricePerVote Price of 1 vote per second (in wei)
+  /// @param maxDuration Maximum duration (in weeks) that a Boost can last when taken from this Offer
+  /// @param expiryTime Timestamp when this Offer is not longer valid
+  /// @param minPerc Minimum percent of users voting token balance to buy for a Boost (in BPS)
+  /// @param maxPerc Maximum percent of users total voting token balance available to delegate (in BPS)
+  /// @param useAdvicePrice True to use the advice Price instead of the given pricePerVote
+  function sellBoost(
+    address underlying,
+    uint256 pricePerVote,
+    uint64 maxDuration,
+    uint64 expiryTime,
+    uint16 minPerc,
+    uint16 maxPerc,
+    bool useAdvicePrice
+  ) external onlyOwnerOrGuardian {
+    VeToken storage token = veTokens[underlying];
+    if (token.warden == address(0)) revert TokenNotRegistered();
+
+    IERC20(IWardenBoost(token.warden).delegationBoost()).safeApprove(
```

Comment:
```suggestion
    IERC20(IWardenBoost(token.warden).delegationBoost()).approve(
```

Reply:
Third comment resolved without making the change or explanation? @efecarranza